src/circuit/gadgets/constraints.rs

```diff
@@ -150,38 +150,6 @@ pub(crate) fn add_to_lc<F: PrimeField, CS: ConstraintSystem<F>>(
     Ok(v_lc)
 }
 
-/// If premise is true, enforce `a` fits into 64 bits. It shows a non-deterministic
-/// partial bit decomposition in order to constraint correct behavior.
-pub(crate) fn implies_u64<F: LurkField, CS: ConstraintSystem<F>>(
-    mut cs: CS,
-    premise: &Boolean,
-    a: &AllocatedNum<F>,
-) -> Result<(), SynthesisError> {
-    let mut a_u64 = a.get_value().and_then(|a| a.to_u64()).unwrap_or(0);
-
-    let mut bits: Vec<Boolean> = Vec::with_capacity(64);
-    for i in 0..64 {
-        let b = a_u64 & 1;
-        let b_bool = Boolean::Is(AllocatedBit::alloc(
-            &mut cs.namespace(|| format!("b.{i}")),
-            Some(b == 1),
-        )?);
-        bits.push(b_bool);
-
-        a_u64 /= 2;
-    }
-
-    // premise -> a = sum(bits)
-    implies_pack(
-        &mut cs.namespace(|| "u64 bit decomposition check"),
-        premise,
-        &bits,
-        a,
-    )?;
-
-    Ok(())
-}
-
```

Comment:
Why are we moving everything out of here and into LEM. I thought the goal was to integrate. I would expect the non-LEM-specific constraint-library code to be here.